File: front/src/models/Asteroide.ts
import { EStatutAsteroide } from "@enums/EStatusAsteroide"

export type Asteroide =
{
    id: number,
    idSysteme: number,
    nom: string | null,
    description: string | null,
    statut: EStatutAsteroide,
    positionX: number,
    positionY: number
}

export type AsteroideConnecter = 
{
    idAsteroideA: number,
    idAsteroideB: number,
    distance: string | null
}

export type AsteroideConnecterSupprimerRequete = Omit<AsteroideConnecter, "distance">;
export type AsteroideRequete = Omit<Asteroide, "id">;

File: back/back/Models/Asteroide.cs
using LiteDB;

namespace back.Models;

public sealed class Asteroide
{
     [BsonId]
     public int Id { get; set; }

     [BsonRef]
     public Systeme Systeme { get; set; } = null!;

     public string? Nom { get; set; }
     public string? Description { get; set; }
     public int Statut { get; set; }

     public decimal PositionX { get; set; }
     public decimal PositionY { get; set; }
}
